instrumentation/koala/Appraisals

```diff
@@ -3,7 +3,6 @@
 # Copyright The OpenTelemetry Authors
 #
 # SPDX-License-Identifier: Apache-2.0
-appraise 'koala-3' do
-  gem 'koala', '~> 3.0.0'
```

Comment:
The ~> 3 removal can be discussed, the plan is to release koala v4 which will support faraday 2 (but not faraday 1) so if you want to be able to support both farday 1 and 2 it would be better to remove this restriction

Reply:
I think maybe just excluding that single bad version probably just makes this easier to maintain going forward, something like

```
appraise 'koala' do
  gem 'koala', '!= 3.0.0'
end

wdyt?

Reply:
I think the problem comes from the gemspec and development dependencies as koala ~> 3.0.0, I'll add a commit to this PR removing the restriction here. (I even wonder if it is needed to have it has a its own appraisal)